pkg/tsdb/prometheus/client/client.go

```diff
@@ -13,100 +13,92 @@ import (
 	"github.com/grafana/grafana/pkg/tsdb/prometheus/models"
 )
 
-type doer interface {
-	Do(req *http.Request) (*http.Response, error)
-}
-
 type Client struct {
-	doer    doer
+	http.RoundTripper
 	method  string
 	baseUrl string
 }
 
-func NewClient(d doer, method, baseUrl string) *Client {
-	return &Client{doer: d, method: method, baseUrl: baseUrl}
+func NewClient(rt http.RoundTripper, method, baseUrl string) *Client {
+	return &Client{RoundTripper: rt, method: method, baseUrl: baseUrl}
 }
 
 func (c *Client) QueryRange(ctx context.Context, q *models.Query) (*http.Response, error) {
-	u, err := url.ParseRequestURI(c.baseUrl)
+	u, err := url.Parse("/api/v1/query_range")
```

Comment:
Why is this change here does not seem like it's entirely connected to the other changes.